src/actions/multiCallHooks.js

```diff
@@ -62,14 +62,17 @@ export const useSubmitRegistration = (registrationDependencies) => {
                         .map((data) => data[0]);
 
                     // set enrollment attribute appropriately
-                    classRegistrations = classRegistrations.map((reg) => ({
-                        ...reg,
-                        // true if matching enrollment
-                        "enrollment": currEnrollments
-                            .some(({student, course}) =>
+                    classRegistrations = classRegistrations.map((reg) => {
+                        const matchingEnrollment = currEnrollments
```

Comment:
once @jerrylinew confirms on the enrollments endpoint, update this to use the course filter.